accepted/00076-configuration-channels-visibility.md

```diff
@@ -0,0 +1,201 @@
+
+### `Configuration > Overview`
+Before:
+![Configuration Overview BEFORE](images/configuration-channels-visibility/Configuration_Overview_BEFORE.png)
+
+After:
+![Configuration Overview AFTER](images/configuration-channels-visibility/Configuration_Overview_AFTER.png)
+
+### `Configuration > Channels`
+Before:
+![Configuration Channels BEFORE](images/configuration-channels-visibility/Configuration_Channels_BEFORE.png)
+
+After:
+![Configuration Channels AFTER](images/configuration-channels-visibility/Configuration_Channels_AFTER.png)
+
+### `Configuration > Channels > Details > Overview`
+Before:
+![Configuration Channels Detail Overview BEFORE](images/configuration-channels-visibility/Configuration_Channels_Details_Overview_BEFORE.png)
+
+After:
+![Configuration Channels Detail Overview AFTER](images/configuration-channels-visibility/Configuration_Channels_Details_Overview_AFTER.png)
+
+### `Configuration > Channels > Details > Systems > Inheriting Systems`
+This is a new page where systems are listed only if not subscribed to the channel directly, but inheriting it through Organization or System Group assignment.
+
+Before:
+![Configuration Channels Detail Systems BEFORE](images/configuration-channels-visibility/Configuration_Channels_Details_Systems_BEFORE.png)
+
+After:
+![Configuration Channels Detail Systems AFTER](images/configuration-channels-visibility/Configuration_Channels_Details_Systems_AFTER.png)
+
+
+### `Configuration > Channels > Details > System Groups`
+This page reproduces the same behavior of the `Configuration > Channels > Details > Systems` tab, but for System Groups.
+
+This **new** page is pointed by:
+- the `Channel information` box link of the system groups counting in the `Configuration > Channels > Details > Overview` page: the link will land on the new page at the `Subscribed System Groups` tab
+
+### `Configuration > Channels > System Groups`
+This page represents a complete list of pairs value `Configuration State Channel - System Group`: every value is a link to the specific entity detail (channel or system group).
+
+This **new** page is pointed by:
+- the `Configuration summary` box link of the system groups counting in the `Configuration > Overview` page: the link will land on the new page where all the pairs `Configuration State Channel - System Group` are listed
+- the `Configuration > Channels` list, for every channel entry, from the `Subscribed System Groups` counting link value
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Why should we **not** do this?
+
+No real reason for not enhancing the feature adding the missing visibility.
+
+The concrete drawback is: the solution is adding information to a tech-debt affected architecture, spreading the same information in many places (see how many pages/screenshots). Maintaining should not be complicated, but future refactoring of the feature workflow could be more difficult.
```

Comment:
Is the tech debt you are referring to Struts? How much more Struts will we be adding? It seems to me rewriting every page we need to touch in ReactJS is rewriting a good portion of the UI :-|

Reply:
The tech debt part is about Struts yes, but it touches it directly and indirectly. One reason is the direct way of introducing data into a page: when it is about Struts we need to fetch and inject data at the same time because there is not a real separation of MVC so we will have to edit both the `ActionController` and the `JSP` to render the information required. But all of this implementation we cannot reuse it because in the ReactJS stack we have a better separation of context where we can fetch and serve data at a specified endpoint, then different pages can consume the information independently and reusing such an endpoint. So in the end we are adding the double of the code to maintain because we touch both the stacks for the same thing.
The indirect point instead is the fact that the UX cannot be changed that much (mainly for consistency) due to existing pages and sub-tabs, so we need to keep the same behavior where we present a link to a different page, instead of rethinking to the entire workflow, with possible dynamic interactions inside the same page, by swapping content or adding boxes, popup frame list or so. This is not possible, or at least it does not worth the effort to implement it in the Struts/JSP stack.

From an Usability perspective, rewriting a good portion of the UI in this case would be something good because the UX is quite old and cumbersome (many tabs and sub tabs pages, lots of tables to touch, different places to present partial information), it could be simplified and smoothed in the end. But of course it would require more time and effort to achieve.

Reply:
Probably it can be seen as part of the `## What other designs/options have been considered?` section. Therefore we will tackle it during a bigger refactoring, even if I am not yet sure when dropping the traditional stack will come.